students/mariusz_michalczyk/lesson_08_regex/regex_strip.py

```diff
@@ -0,0 +1,16 @@
+import re
+
+
+def strip_mm(str, replacement=''):
+    if not replacement:
+        rgx_strip = re.compile(r'\s*(\d*\w*)\s*')
```

Comment:
You can't assume the group won't have spaces: `strip('   aaa   aaa   ') == 'aaa   aaa'`. 

One way is to do the same as you did with replacement, but use `\s` (I'm not sure about newline characters... maybe `[\n\s]`) instead of `[{0}]`.

Another is to actually match everything in the group, but not eagerly, because that would catch the spaces at the end too, i.e. `(.*?)` (the `*?` stands for match many, but as little as possible).

Reply:
written test prove that it works.